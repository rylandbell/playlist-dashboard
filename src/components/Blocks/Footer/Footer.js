import React from 'react';
import './Footer.css';

const Footer = () =>
  <div className="footer">
    <ul className="list-inline">
      <li>
        <small>
          View project on&nbsp;
          <a
            href="https://github.com/rylandbell/spotify-features-filter"
            target="_blank"
            rel="noopener noreferrer"
          >
            GitHub
          </a>
        </small>
      </li>
      <li>
        <small>
          Audio features data by&nbsp;
          <a
            href="https://spotify.com"
            target="_blank"
            rel="noopener noreferrer"
          >
            Spotify
          </a>
        </small>
      </li>
      <li>
        <small>
          Icons by&nbsp;
          <a
            href="https://icons8.com"
            target="_blank"
            rel="noopener noreferrer"
          >
            Icons8
          </a>
        </small>
      </li>
    </ul>
  </div>;

export default Footer;
