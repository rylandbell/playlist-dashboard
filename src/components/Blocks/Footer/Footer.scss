@import 'variables';

.footer {
  background-color: rgb(40,40,40);
  color: $text-light;
  flex: 1 1 auto;
  margin: 0;
  position: absolute;
  bottom: 0;
  padding: 3px 10px;
  
  ul {
    margin: 0;
    display: inline-flex;
    flex-direction: row;
    
    li {
      // text-align: center;
      flex: 1 1 auto;
      border: 1px dashed red;
    }
  }
}