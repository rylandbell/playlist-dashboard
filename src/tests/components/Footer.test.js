import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Footer from "../../components/blocks/Footer/Footer";

describe("Footer component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Footer />, div);
  });

  it("shallow render matches snapshot", () => {
    const tree = toJson(shallow(<Footer />));
    expect(tree).toMatchSnapshot();
  });
});
