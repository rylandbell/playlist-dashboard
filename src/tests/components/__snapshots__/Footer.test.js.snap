// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component shallow render matches snapshot 1`] = `
<div
  className="footer"
>
  <ul
    className="list-inline"
  >
    <li>
      <small>
        View project on 
        <a
          href="https://github.com/rylandbell/playlist-dashboard"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </small>
    </li>
    <li>
      <small>
        Audio features data by 
        <a
          href="https://spotify.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          Spotify
        </a>
      </small>
    </li>
    <li>
      <small>
        Icons by 
        <a
          href="https://icons8.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          Icons8
        </a>
      </small>
    </li>
  </ul>
</div>
`;
